src/legacy/index.ts

```diff
@@ -103,7 +116,14 @@ async function buildGraph(
       version: depTree.version,
     };
     pkgNodeId = getNodeId(pkg.name, pkg.version, treeHash);
-    builder.addPkgNode(pkg, pkgNodeId);
+
+    const nodeInfo: types.NodeInfo = {};
+
+    if (depTree.versionProvenance) {
+      nodeInfo.versionProvenance = depTree.versionProvenance;
```

Comment:
should we be worried about hanging on to a reference here? potential source of bugs...